### DISCUSSION TITLE
nextjs-jest-rust should support dynamic loading

### DISCUSSION THREAD

**Discussion by to-long** [vercel/next.js#34606] [Ideas]
### Describe the feature you'd like to request

I want to test dynamic component, but I only this way: config .babelrc
[https://stackoverflow.com/questions/46906948/how-to-unit-test-next-js-dynamic-components](https://stackoverflow.com/questions/46906948/how-to-unit-test-next-js-dynamic-components)
But with .babelrc enabled, swc cannot run 
![image](https://user-images.githubusercontent.com/8638967/154839377-e629ced6-15c1-46d0-a515-31a955e8127e.png)

### Describe the solution you'd like

please support to load dynamic component in jest with rust config

### Describe alternatives you've considered

n/a

**Comment by balazsorban44** [comment]
Hi, this should already be supported. I created this simple example:


```js
// pages/index.js

import dynamic from "next/dynamic"

const Component = dynamic(() => import("../component"), {
  loading: () => <span>Loading...</span>,
  ssr: false,
})

export default function Home() {
  return <Component />
}
```


```js
// component.js
export default function Component() {
  return <span>Loaded</span>
}
```

```js
// __tests__/index.test.js

import { render, screen } from "@testing-library/react"
import Home from "@/pages/index"

describe("Home", () => {
  it("renders dynamic component", async () => {
    render(<Home />)

    expect(screen.getByText("Loading...")).toBeInTheDocument()

    expect(await screen.findByText("Loaded")).toBeInTheDocument()
  })
})

```

**Reply by to-long** [reply]
After I upgraded from nextjs 12.0.9 -> 12.1.0 , this issue was resolved
Thanks for confirm it, I think we can close this issue

**Reply by Zebak2021** [reply]
Thanks
